### PR TITLE
wrote exists function - still needs to be tested

### PR DESCRIPTION
**Pull request by SammyRobensParadise**

### CODE REVIEW

File: Lab3/binary_search_tree.cpp [hunk 1]
```diff
@@ -54,47 +54,64 @@ unsigned int BinarySearchTree::height() const {
...
-    return false;
+// TODO @Sammy
+
+
+bool BinarySearchTree::exists(BinarySearchTree::TaskItem val) const {
+    if (root == NULL) {
+        cerr << "Empty tree\n";
+        return false;
+    } else if (size == 1) {
+        return val == *root ? true : false;
```
Comment (maryrsmcphee) [inline Lab3/binary_search_tree.cpp]: I don't think we need the ` ? true : false; `

File: Lab3/binary_search_tree.cpp [hunk 2]
```diff
@@ -54,47 +54,64 @@ unsigned int BinarySearchTree::height() const {
...
+    }else{
+        if(*root == val){
+            true;
```
Comment (maryrsmcphee) [inline Lab3/binary_search_tree.cpp]: Should this be `return true;`?

**Review by maryrsmcphee — APPROVED** [review]

File: Lab3/binary_search_tree.cpp (hunk 1 above)
Comment (SammyRobensParadise) [reply Lab3/binary_search_tree.cpp]: if its of size one and it equals the root then return true, else return false?

File: Lab3/binary_search_tree.cpp (hunk 2 above)
Comment (SammyRobensParadise) [reply Lab3/binary_search_tree.cpp]: oh yeah lol

**Comment by SammyRobensParadise** [comment]
@maryrsmcphee made some change lmk what you think

File: Lab3/binary_search_tree.cpp (hunk 1 above)
Comment (maryrsmcphee) [reply Lab3/binary_search_tree.cpp]: oh like it's not important just `return val == * root;` means the same thing as `return val == *root ? true : false`

**Comment by SammyRobensParadise** [comment]
@maryrsmcphee  you may also want to make it so that you need another approve when you push more code in settings

File: Lab3/binary_search_tree.cpp (hunk 2 above)
Comment (SammyRobensParadise) [reply Lab3/binary_search_tree.cpp]: fixed in latest commit

File: Lab3/binary_search_tree.cpp (hunk 1 above)
Comment (SammyRobensParadise) [reply Lab3/binary_search_tree.cpp]: okay ya thats cleaner ill do that

Comment (SammyRobensParadise) [reply Lab3/binary_search_tree.cpp]: fixed in latest commit

**Review by maryrsmcphee — APPROVED** [review]
Looks good!

**Comment by SammyRobensParadise** [comment]
okay probs good to require another approval after each commit fyi

**Comment by maryrsmcphee** [comment]
yep just fixed

**Pull request merged**